### ISSUE TITLE
Create default wrapped clients only if necessary also for S3AsyncEncryptionClient

### ISSUE THREAD

**Issue by JoergHeinicke5005** [aws/amazon-s3-encryption-client-java#207]
### Problem:
If the region is not set in the environment, instantiation of S3Client fails. Therefore, when setting up S3Client, we also set the region on the client. That works with S3EncryptionClient:

```
S3EncryptionClient.builder()
                .wrappedClient(S3Client.builder().region(EU_CENTRAL_1).build())
                .wrappedAsyncClient(S3AsyncClient.builder().region(EU_CENTRAL_1).build())
                .build();
```

Unfortunately, it does not with the S3AsyncEncryptionClient as its Builder instantiates a default wrapped client without a region set.

```
S3AsyncEncryptionClient.builder()
                .wrappedClient(S3AsyncClient.builder().region(EU_CENTRAL_1).build())
                .build();
```

### Solution:

Apply the same fix as for S3EncryptionClient, commit 285eab68fdc2468e678d3b745d6502a9584752e4.

**Comment by JoergHeinicke5005** [comment]
Just noticed it had just been fixed last week in #187.

**Comment by JoergHeinicke5005** [comment]
Any chance to release new version soon with the fix?
